### ISSUE TITLE
[Issue]: Allow items with tags breaks media access and playback when movies placed in subfolders

### ISSUE THREAD

**Issue by 46o60** [jellyfin/jellyfin#11869]
### Please describe your bug

Thank you for your great application! I am new user to Jellyfin, just recently switched from Emby after several years of usage. So far I am happy but I encountered an issue that I want to report.

I am setting up access for my child and I want to control through Parental Control "Allow items with tags" option which movies they can access and watch. I want to use whitelist approach to gradually allow more and more movies while having strict control on which movies I allow.

First thing I do is add a custom tag to a movie:

![Screenshot 2024-05-28 at 23 28 41](https://github.com/jellyfin/jellyfin/assets/29371699/79784bb4-b4ad-424b-9954-19c05367b136)

After that, I go to their user and configure the Parental Control, Allow items with tags and set that same tag value there:

![Screenshot 2024-05-28 at 23 48 04](https://github.com/jellyfin/jellyfin/assets/29371699/d17cf773-dfb2-4f5f-a268-75cfb2fae335)

When I log in to their account I correctly see the library and only this movie in that library that has the configured tag:

![Screenshot 2024-05-28 at 23 29 29](https://github.com/jellyfin/jellyfin/assets/29371699/77595555-73f8-4bfc-a49b-e7238fa40c83)

However, when I click on that movie the page never loads with the movie info:

![Screenshot 2024-05-28 at 23 30 14](https://github.com/jellyfin/jellyfin/assets/29371699/e62d2ffa-abc3-472e-a84e-92ddbc19acb1)

The same errors in the console are shown if I try to play the movie directly from the library movie list or try to change the Watched status.

If I go back and clear the Allow items with tags settings, the movie page loads as expected, and I can play the video without any further issues:

![Screenshot 2024-05-28 at 23 35 42](https://github.com/jellyfin/jellyfin/assets/29371699/6e2554df-0554-404a-b4cc-840208fa4523)

I noticed this initially on 10.9.2 but since then I upgraded to 10.9.3 and still the same behaviour. This deployment is an Debian LXC container fully updated running on Proxmox host. The same issue manifests both in Chrome as well as in Android Jellyfin app. Other than this, all clients and server work perfectly fine.

```
root@jellyfin:~# uname -a
Linux jellyfin 6.8.4-2-pve #1 SMP PREEMPT_DYNAMIC PMX 6.8.4-2 (2024-04-10T17:36Z) x86_64 GNU/Linux
root@jellyfin:~# cat /etc/os-release 
PRETTY_NAME="Debian GNU/Linux 12 (bookworm)"
NAME="Debian GNU/Linux"
VERSION_ID="12"
VERSION="12 (bookworm)"
VERSION_CODENAME=bookworm
ID=debian
HOME_URL="https://www.debian.org/"
SUPPORT_URL="https://www.debian.org/support"
BUG_REPORT_URL="https://bugs.debian.org/"
root@jellyfin:~# jellyfin --version
Jellyfin.Server 10.9.3.0
```

Based on the behaviour it seems related to how tags are being handled on the server side. I saw in existing issues #11407 but that seems different than what I am experiencing.

### Reproduction Steps

See description above.

### Jellyfin Version

10.9.0

### Environment

```markdown
- OS: Debian GNU/Linux 12 (bookworm)
- Linux Kernel: 6.8.4-2-pve
- Virtualization: Proxmox
- Clients: Android Jellyfin 
- Browser: Chrome 125.0.6422.77
- FFmpeg Version: ffmpeg version 6.0.1-Jellyfin
- Playback Method: N/A
- Hardware Acceleration: no
- GPU Model: N/A
- Plugins: none
- Reverse Proxy: no
- Base URL: N/A
- Networking: N/A
- Storage: N/A
```

### Jellyfin logs

```shell
No relevant logs found.
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by gnattu** [comment]
I can't reproduce this. Can you share more info?

Like do you have any other parental settings for the account and how is your library organized?

**Comment by gnattu** [comment]
A quick way to test and rule out other problem is, create a new user and a new library, put only one movie in that library, and assign a tag to that, and set the new user's allowed tag to that tag, and check if that user can see and play this movie correctly.

**Comment by nielsvanvelzen** [comment]
Do you have plugins installed for intros?

**Comment by 46o60** [comment]
@nielsvanvelzen I do not have any plugins installed apart from the default ones.

![image](https://github.com/jellyfin/jellyfin/assets/29371699/821c0c22-826a-4f73-9777-96dd827a15b2)

@gnattu Thank you for trying to reproduce. To answer your questions:

1. No I do not have any other parental settings configured. I configured whitelist of tags.
2. I do not think there is anything special with my library (but I am maybe wrong, see below)
3. I have created a new user and a new library and the issue seems to disappear then. However...

While I was setting up new library I have noticed one thing. My library on the file system looks like this:

```
RootLibraryDir
|_ Movie 1 Name (YYYY)\
|_ Movie 2 Name (YYYY)\
|_ Directory Name
  \__ Movie 3 Name (YYYY)\
```

Seems that the original issue I described is valid only for movies that are located in subdirectories, such as Movie 3 in the above example. I have replicated the same in the newly created library with the new user.

This is the structure I have in my testing library (I snipped out artwork files):

```
root@jellyfin:~# ls -lR /mnt/moviestemp/
/mnt/moviestemp/:
total 0
drwxrwx--- 2 root host_shares 0 May 29 21:11 LibraryTemp

/mnt/moviestemp/LibraryTemp:
total 0
drwxrwx--- 2 root host_shares 0 May 29 21:12  Cartoons
drwxrwx--- 2 root host_shares 0 May 28 23:25 'Daffy Duck'\''s Movie - Fantastic Island (1983)'

/mnt/moviestemp/LibraryTemp/Cartoons:
total 0
drwxrwx--- 2 root host_shares 0 May 27 23:38 'Cinderella II - Dreams Come True (2002)'

'/mnt/moviestemp/LibraryTemp/Cartoons/Cinderella II - Dreams Come True (2002)':
total 789679
-rwxrwx--- 1 root host_shares 809584434 Oct  1  2008 'Cinderella II - Dreams Come True.Hr.avi'
[...SNIP...]

'/mnt/moviestemp/LibraryTemp/Daffy Duck'\''s Movie - Fantastic Island (1983)':
total 696767
-rwxrwx--- 1 root host_shares 713240843 Jan 15  2017 "Daffy Duck's Movie - Fantastic Island 1983.mp4"
[...SNIP...]
```

In the above library setup, bot Daffy and Cinderella movies have a custom tag `TagTest` but Cinderella is in a subdirectory `Cartoons`. A user `UserTest` has parental control configured to allow only movies with that same `TagTest` tag. Library `MoviesTest2` has folder path `/mnt/moviestemp/LibraryTemp`:

![Screenshot 2024-05-29 at 21 22 01](https://github.com/jellyfin/jellyfin/assets/29371699/607ff58a-7643-4a7c-995e-7f60e8a0d567)

As `UserTest` opening the `MoviesTest2` library shows both movies:

![Screenshot 2024-05-29 at 21 22 32](https://github.com/jellyfin/jellyfin/assets/29371699/0c17c521-a56a-4ebe-8e7c-1d52737b337f)

When I click on Daffy the movie details page opens normally and I can play it:

![Screenshot 2024-05-29 at 21 23 37](https://github.com/jellyfin/jellyfin/assets/29371699/7d11eb49-8cf7-4240-abbd-dd711aa3845e)

However, when I click on Cinderella I get the same behaviour as originally described - 404 status codes and movie details page never loading:

![Screenshot 2024-05-29 at 21 24 31](https://github.com/jellyfin/jellyfin/assets/29371699/6a59b566-278c-48da-aec2-d68e10629193)

I did not see any warnings to organise file system for library in a flat way... I have subdirectories to have a bit more organisation, for example keeping all cartoon movies in one sub directory in the library. And interestingly, if I remove parental control for whitelisted tags I can open normally both movies. So directory structure alone is not the problem but specifically tag whitelisting and subdirectories are not playing well together.

Clicking on Cinderella movie after I removed parental control with whitelisted tags loads also that movie details page without any issues and can be played:

![image](https://github.com/jellyfin/jellyfin/assets/29371699/26b9e628-6e80-401f-b959-9f95958fa872)

Hopefully you will be able to reproduce this in your own environment. Is there anything else I can do to provide more detailed logs?

**Comment by 46o60** [comment]
I have just tried to move Daffy inside `Cartoons` folder and Cinderella outside of `Cartoons` folder and the behaviour is reversed, not Cinderella opens without any issues by Daffy has 404s in the console. I think this is strong indication that subfolders are indeed somehow connected to this issue. Again, removing tag whitelist parental control allows opening and playing both movies so subfolders on their own are not the issue.

**Comment by elfalem** [comment]
I think I have a similar issue but with music. I also think I know the cause of the issue.

In my case, I tagged a music album, and specified that tag to a user. Logging in as that user, the album is visible, but clicking on it to play/view contents shows a spinner. Requests such as `http://localhost:8096/Users/b0fffac5f0774ae4b16cf216c339b6cc/Items/3a3ce2f87f9d47ae621334723280a7f2` are initiated in browser console, but the return status is 404 (just like OP) and the web UI shows a spinner. If I remove the allowed tag restriction, the user is able to see all items in the library.

My media follows a conventional directory structure of `library-root/artist/album/track.flac`.

I haven't debugged the code, but just reading through it, I think the 404 is being returned [here](https://github.com/jellyfin/jellyfin/blob/1f5ce4e4a25f550aac150e713c76d51f722a4651/Jellyfin.Api/Controllers/UserLibraryController.cs#L90) because the library manager is not able to get the item by id. Digging deeper, [here](https://github.com/jellyfin/jellyfin/blob/master/MediaBrowser.Controller/Entities/BaseItem.cs#L1306) the code checks that all parents (I'm guessing meaning all ancestors) are visible to the user. One of the [visibility checks](https://github.com/jellyfin/jellyfin/blob/master/MediaBrowser.Controller/Entities/BaseItem.cs#L1534) is for whether the item is visible through tags or not.

In other words, the code checks that all items in the hierarchy up to the library have a whitelisted tag. In my case the album had the tag because that's where I applied it, the tracks have the tag because they inherited it from the album, but the artist directory doesn't have it so it fails the check.

When I added the tag to the artist directory through the metadata manager, I was able to see and play the tracks.

For @46o60 's case, I'm guessing tagging the `Cartoons` folder with `TagTest` would make it work.

**Comment by elfalem** [comment]
I have created a PR that should fix this. I have tested it for both the music scenario as well as the movie in subfolder scenario.

**Comment by elfalem** [comment]
@LiseDerg based on your description, it does sound like the same issue.

**Comment by LiseDerg** [comment]
@elfalem okey thanks,
well then we just have to wait for it to be solved or the changes to be applied so that it works in any situation to allow with tag.

**Comment by elfalem** [comment]
@LiseDerg , @46o60  - this has been fixed with the 10.10.6 release.

**Comment by LiseDerg** [comment]
> @LiseDerg , @46o60  - this has been fixed with the 10.10.6 release.

thank you for the notice.

and thanks to everyone for keeping jellyfin running.
